sdk/core/core-rest-pipeline/src/policies/tracingPolicy.ts

```diff
@@ -93,11 +94,19 @@ export function tracingPolicy(options: TracingPolicyOptions = {}): PipelinePolic
         if (serviceRequestId) {
           span.setAttribute("serviceRequestId", serviceRequestId);
         }
-        span.end();
+        span.setStatus({
+          code: SpanStatusCode.OK
+        });
         return response;
       } catch (err) {
-        span.end();
+        span.setStatus({
```

Comment:
I will be very honest here - I was able to e2e test the core-http changes using KeyVault (installing the built package) but here I was just guessing based on the types.

I would much prefer to test this code path e2e - would someone be able to help me do that? The samples are empty 😞